src/components/footerAppBar/FooterAppBar.tsx

```diff
@@ -2,33 +2,84 @@ import React from 'react'
 import BottomNavigation from '@mui/material/BottomNavigation'
 import BottomNavigationAction from '@mui/material/BottomNavigationAction'
 import Paper from '@mui/material/Paper'
+import { Link } from 'react-router-dom'
+import { useActivePage } from '../../context/activePageContext'
+
+const FooterAppBar: React.FC = () => {
+  const { activePage, setNewActivePage } = useActivePage()
 
-const FooterAppBar = () => {
   return (
     <Paper
       sx={{ position: 'fixed', bottom: 0, left: 0, right: 0 }}
       elevation={3}
     >
-      <BottomNavigation>
+      <BottomNavigation
+        value={activePage}
+        onChange={(event, newValue) => setNewActivePage(newValue)}
+      >
         <BottomNavigationAction
-          label="Recents"
-          icon={<img src="/img/nearBy/near_me.svg" />}
+          value="nearme"
+          icon={
+            <Link to="/nearMe" style={{ textDecoration: 'none' }}>
+              {activePage === 'nearme' ? (
+                <img src="/img/navigationIcons/near_me_red.svg" alt="near_me" />
+              ) : (
+                <img src="/img/navigationIcons/near_me.svg" alt="near_me" />
+              )}
+            </Link>
+          }
         />
         <BottomNavigationAction
-          label="Favorites"
-          icon={<img src="img/nearBy/like_you.svg" />}
+          value="wholikedyou"
+          icon={
+            <Link to="/whoLikedYou" style={{ textDecoration: 'none' }}>
```

Comment:
Same here, it's hard to read code

Reply:
Refactored components Header.tsx and FooterAppBar.tsx to get rid of conditionals. Created the following helpers: navigationConfig.ts, navigationRenderer.tsx, renderIcon.ts.